### PR TITLE
Refactor group collection structure to store summary

### PR DESCRIPTION
**Pull request by cp-amisha-i**
## Summary by CodeRabbit

- **New Features**
	- Improved handling of total amounts in group financial summaries, enhancing clarity and accuracy.
	- Streamlined calculation methods for group spending and payments, resulting in a more cohesive user experience.
	- Enhanced visual representation of payment amounts, improving the overall UI consistency.

- **Bug Fixes**
	- Resolved issues with data fetching and display, ensuring accurate representation of group financial data.
	- Improved filtering logic for expenses, leading to more relevant data presentation based on user selection.

### CODE REVIEW

File: Data/Data/Store/ExpenseStore.swift [hunk 1]
```diff
@@ -187,4 +187,49 @@ public class ExpenseStore: ObservableObject {
...
     }
+
+    func fetchCurrentMonthExpensesBy(groupId: String) -> AnyPublisher<[Expense], ServiceError> {
+        Future { [weak self] promise in
+            guard let self else {
+                promise(.failure(.unexpectedError))
+                return
+            }
+
+            let calendar = Calendar.current
```
Comment (cp-amisha-i) [inline Data/Data/Store/ExpenseStore.swift:198]: This type of calculated value should pass from the repository whenever possible.

File: Splito/UI/Home/Groups/Group/GroupHomeViewModel.swift [hunk 2]
```diff
@@ -350,9 +381,14 @@ extension GroupHomeViewModel {
     }
 
     func dismissShowSettleUpSheet() {
+        fetchLatestExpenses()
         showToastFor(toast: .init(type: .success, title: "Success", message: "Payment made successfully"))
         showSettleUpSheet = false
     }
+
+    func dismissSheetCallback() {
+        fetchLatestExpenses()
```
Comment (cp-amisha-i) [inline Splito/UI/Home/Groups/Group/GroupHomeViewModel.swift:390]: This is not the right way to observe real-time updates, let's not manually fetch the updated data with any change.

File: Splito/UI/Home/Groups/Group/GroupHomeViewModel.swift [hunk 3]
```diff
@@ -179,12 +185,10 @@ class GroupHomeViewModel: BaseViewModel, ObservableObject {
     private func fetchGroupBalance() {
         guard let userId = preference.user?.id, let group else { return }
 
-        memberOwingAmount = [:]
-        overallOwingAmount = 0.0
```
Comment (cp-amisha-i) [inline Splito/UI/Home/Groups/Group/GroupHomeViewModel.swift:165]: Let's please not remove this.

Comment (cp-nirali-s) [reply Splito/UI/Home/Groups/Group/GroupHomeViewModel.swift:165]: But what is the reason for keep it ?

File: Splito/UI/Home/Groups/Group/GroupHomeViewModel.swift (hunk 2 above)
Comment (cp-nirali-s) [reply Splito/UI/Home/Groups/Group/GroupHomeViewModel.swift:375]: But when sheet is dismissed then group home view model's overall amount and member owing amount will not update automatically then how we will update this variable ?

File: Splito/UI/Home/Groups/CalculateExpensesFunctions.swift [hunk 4]
```diff
@@ -137,6 +137,11 @@ public func getUpdatedMemberBalanceFor(transaction: Transactions, group: Groups,
         memberBalance.append(GroupMemberBalance(id: receiverId, balance: amount))
     }
 
+    let epsilon = 1e-10
```
Comment (cp-amisha-i) [inline Splito/UI/Home/Groups/CalculateExpensesFunctions.swift:140]: This is for what? I'm not getting it.

Comment (cp-nirali-s) [reply Splito/UI/Home/Groups/CalculateExpensesFunctions.swift:140]: This is for removing small floating-point number you're seeing (e.g., -2.842170943040401e-14) are due to the way computers handle floating-point arithmetic. These numbers are common when dealing with floating-point numbers and result from the imprecision in how these numbers are stored and calculated in binary.

**Pull request merged**